lib/set-script.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const log = require('npmlog')
+const usageUtil = require('./utils/usage.js')
+const { localPrefix } = require('./npm.js')
+const fs = require('fs')
+const usage = usageUtil('set-script', 'npm set-script [<script>] [<command>]')
+const completion = require('./utils/completion/none.js')
+const parseJSON = require('json-parse-even-better-errors')
+const rpj = require('read-package-json-fast')
+
+const cmd = (args, cb) => set(args).then(() => cb()).catch(cb)
+
+const set = async function (args) {
+  if (process.env.npm_lifecycle_event === 'postinstall')
+    throw new Error('Scripts can\'t set from the postinstall script')
```

Comment:
```suggestion
    throw new Error('Scripts can’t set from the postinstall script')
```
minor nit, since straight quotes are typographically incorrect in prose :-)